cli/cmds/utils

```diff
@@ -0,0 +1,589 @@
+    local chain_id=$(printf "%s\n" "${SUPPORTED_CHAINS[@]}" | grep "$chain" | cut -d':' -f1)
+    echo "$chain_id:$chain"
+}
+
+set_rpc_url() {
+    case $1 in
+        1) RPC_URL="https://cloudflare-eth.com" ;; # Ethereum
+        137) RPC_URL="https://polygon-rpc.com" ;; # Polygon
+        8453) RPC_URL="https://mainnet.base.org" ;; # Base
+        42161) RPC_URL="https://arb1.arbitrum.io/rpc" ;; # Arbitrum
+        1329) RPC_URL="https://evm-rpc.sei-apis.com" ;; # Sei
+        33139) RPC_URL="https://curtis.rpc.caldera.xyz/http" ;; # ApeChain
+        *) echo "Unsupported chain id"; exit 1 ;;
+    esac
+
+    export RPC_URL
+}
+
+generate_merkle_root() {
+    generateMerkleRootPath="$BASE_DIR/../../scripts/utils/generateMerkleRoot.ts"
+
+    if [[ ! -f "$generateMerkleRootPath" ]]; then
+        echo "Error: generateMerkleRoot.ts not found at $generateMerkleRootPath"
+        return 1
+    fi
+
+    result=$(npx ts-node "$generateMerkleRootPath" "$1" false)
+    echo "$result" | sed -n 's/^Merkle Root: *//p'
+}
+
+load_stages_json() {
+    local stages_file="$1"
+    if [[ ! -f "$stages_file" ]]; then
+        echo "Error: Stages file not found: $stages_file"
+        return 1
+    fi
+
+    # Extract whitelistPath
+    local whitelistPaths
+    whitelistPaths=$(jq -r '.[].whitelistPath' "$stages_file")
+
+    merkle_roots=()
+    for whitelistPath in $whitelistPaths; do
+        if [[ -z "$whitelistPath" || "$whitelistPath" == "null" ]]; then
+            merkle_roots+=("$DEFAULT_MERKLE_ROOT")
+        else
+            echo "Cleaning whitelist files and generating merkle roots for $whitelistPath..."
+            result=$(generate_merkle_root "$BASE_DIR/../collections/$whitelistPath")
+            if [[ $? -ne 0 ]]; then
+                echo "Error: Failed to generate merkle root"
+                return 1
+            fi
+            merkle_roots+=("$result")
+        fi
+    done
+    json_array=$(printf '%s\n' "${merkle_roots[@]}" | jq -R . | jq -s .)
+
+    # Map the stages data from the JSON file, and replace the merkle root with the generated one
+    stages_data=$(jq -c --argjson merkle_roots "$json_array" '
+        to_entries | map(
+            [
+                (.value.price | tonumber * 1e18 | floor),
+                (.value.mintFee | tonumber * 1e18 | floor),
+                (.value.walletLimit | tonumber),
+                ($merkle_roots[.key] // empty),
+                (.value.maxStageSupply // 0 | tonumber),
+                (.value.startDate | fromdateiso8601 | tonumber),
+                (.value.endDate | fromdateiso8601 | tonumber)
+            ]
+        )
+    ' "$stages_file")
+
+
+    if [[ $? -ne 0 ]]; then
+        echo "Error: Failed to parse stages JSON"
+        return 1
+    fi
+
+    stages_data=$(echo "$stages_data" | sed 's/"//g') # remove any quotes
+
+    # Replace inner square brackets with parentheses, but keep the outer brackets
```

Comment:
This is another example. It's very hard to understand and modify this transforming logic.